meeting.js

```diff
@@ -0,0 +1,330 @@
+// 1) Access token and SIP URI (Hard Coded).
+const accessToken = "";
+const sipAddress = "";
+
+// 2) Setting Media Variables.
+const joinMeetingButton = document.getElementById('joinMeetingButton');
+const leaveMeetingButton = document.getElementById('leaveMeetingButton');
+const microphoneButton = document.getElementById('microphone');
+const videoButton = document.getElementById('video');
+const RemoteVideo = document.getElementById('remoteVideo');
+const RemoteAudio = document.getElementById('remoteAudio');
+const localVideo = document.getElementById('localVideo');
+const shareButton = document.getElementById('share');
+const bnrButton = document.getElementById('bnr');
+const vbgButton = document.getElementById('vbg');
+
+// 3) Setting Initiated Variables to Null.
+let createdMeeting = null;
+let localStream = null;
+let isMuted = true;
+let isVideoStarted = true;
+let vbgEffect = false;
+
+// 4) Webex Initialize.
+const webex = window.Webex.init({
+  credentials: {
+    access_token: accessToken
+  }
+});
+
+// 5) Registering Webex Device
+webex.meetings.register()
+  .catch((err) => {
+    console.error('Error registering Webex device:', err);
+    alert('Error registering Webex device. Check console for details.');
+  });
+
+// 6) CreateMeeting Function.
+async function createMeeting() {
+  try {
+    const meeting = await webex.meetings.create(sipAddress);
+    createdMeeting = meeting;
+  } catch (error) {
+    console.error('Error creating meeting:', error);
+    throw error;
+  }
+}
+
+// 7) Set Media Listeners(Remote Streams)
+function setMediaListeners() {
+  if (!createdMeeting) {
+    console.error('No meeting available to set media listeners.');
+    return;
+  }
+  createdMeeting.on('media:ready', (media) => {
+    console.log('Media ready', media);
+    switch (media.type) {
+      case 'remoteVideo':
+        RemoteVideo.srcObject = media.stream;
+        break;
+      case 'remoteAudio':
+        RemoteAudio.srcObject = media.stream;
+        RemoteAudio.play();
+        break;
+      default:
+        console.error('Unknown media type:', media.type);
+    }
+  });
+  createdMeeting.on('media:stopped', (media) => {
+    switch (media.type) {
+      case 'remoteVideo':
+        RemoteVideo.srcObject = null;
+        break;
+      case 'remoteAudio':
+        RemoteAudio.srcObject = null;
+        break;
+      default:
+        console.error('Unknown media type:', media.type);
+    }
+  });
+}
+
+// 8) Get Local Streams Function.
+async function getLocalStreams() {
+  try {
+    const microphoneStream = await webex.meetings.mediaHelpers.createMicrophoneStream({
+      echoCancellation: true,
+      noiseSuppression: true,
+    });
+
+    const cameraStream = await webex.meetings.mediaHelpers.createCameraStream({ width: 640, height: 480 });

```

Comment:
Why are we setting specific width and height here?

Reply:
These are the constraits to give the video quality of the localvideo.So we give the constraits for better video quality

Reply:
In the documentation also the have use the constraits like width and height for better quality of local Video Image

Reply:
What happens if we do not give this width and height? I think they are optional. Do we really need to constrain it with a hardcoded resolution here?

Reply:
The resolution will change when we give constraints So I had given Constraints That was the ratio For 1080 p 720p like that . Refer the Documentation and also Sample App Reference uses constraints

Reply:
Updated